jdaviz/core/loaders/importers/spectrum_list/spectrum_list.py

```diff
@@ -23,6 +40,89 @@ def __init__(self, *args, **kwargs):
         else:
             self.data_label_default = '1D Spectrum'
 
+        self.previous_data_label_messages = []
+
+        if self.is_valid:
+            # If the resolver format is set to "1D Spectrum List", then we
+            # only enable the import button if at least one spectrum is selected.
+            # This is to avoid a user accidentally importing all spectra (API default)
+            # when none are selected
+            self.resolver.observe(self._on_format_selected_change, names='format_selected')
+            # Separately observe changes to the selected spectra
+            self.observe(self._on_spectra_selected_change, names='spectra_selected')
+
+            exposures = []
+            spectra_options = []
+
+            if isinstance(self.input, Spectrum):
+                speclist_input = SpectrumList(self.input_to_list_of_spec(self.input))
+            else:
+                speclist_input = self.input
+
+            for index, spec in enumerate(speclist_input):
+                if self.is_wfssmulti(spec):
+                    # ver, name are stand-ins for exposure and source_id
+                    # ver == exposure, name == source_id
+                    ver, name = self._extract_exposure_sourceid(spec)
+                    exposure_label = f"Exposure {ver}"
+                    exposures.append(exposure_label)
+
+                    label = f"{exposure_label}, Source ID: {name}"
+                    # Flipping the two from the variable naming convention
+                    name_ver = f"{ver}_{name}"
+                    suffix = f"EXP-{ver}_ID-{name}"
+
+                else:
+                    name_ver = index
+                    name = index
+                    ver = index
+                    label = f"1D Spectrum at file index: {index}"
```

Comment:
This may not actually be a file input though
```suggestion
                    label = f"1D Spectrum at index: {index}"
```